Makefile

```diff
@@ -2209,6 +2229,31 @@ endif
 	$(MAKE) rocksdbjavastatic_deps
 	$(MAKE) rocksdbjavastatic_libobjects
 	$(MAKE) rocksdbjavastatic_javalib
+	$(MAKE) rocksdbjavastatic_jar
+
+rocksdbjavastaticosx: rocksdbjavastaticosx_archs
+	mv java/target/librocksdbjni-osx-x86_64.jnilib java/target/librocksdbjni-osx.jnilib
+	mv java/target/librocksdbjni-osx-arm64.jnilib java/target/librocksdbjni-osx-aarch64.jnilib
+
+rocksdbjavastaticosx_ub: rocksdbjavastaticosx_archs
+	lipo -create -output ./java/target/$(ROCKSDBJNILIB) java/target/librocksdbjni-osx-x86_64.jnilib java/target/librocksdbjni-osx-arm64.jnilib
+	$(MAKE) rocksdbjavastatic_jar
+
+rocksdbjavastaticosx_archs: 
+	$(MAKE) clean-ext-libraries-bin 
+	$(MAKE) clean-rocks
+	$(MAKE) rocksdbjavastaticosx_arch_x86_64
+	$(MAKE) clean-ext-libraries-bin 
+	$(MAKE) clean-rocks
+	$(MAKE) rocksdbjavastaticosx_arch_arm64
+
+rocksdbjavastaticosx_arch_%:
+ifeq ($(JAVA_HOME),)
+	$(error JAVA_HOME is not set)
+endif
+	CXXFLAGS="$(CXXFLAGS)" ARCHFLAG="-arch $*" $(MAKE) rocksdbjavastatic_deps
```

Comment:
Should the clean-* rule lines from above (2243-2244) be moved here instead?  If a user invoked this rule, they could get flaky results.

Reply:
Updated